lib/Doctrine/Collections/ArrayCollection.php

```diff
@@ -0,0 +1,455 @@
+<?php
+
+namespace Doctrine\Collections;
+
+use ArrayIterator;
+use Closure;
+use Doctrine\Collections\Expr\ClosureExpressionVisitor;
+use Doctrine\Common\Collections\Collection as OldCollection;
```

Comment:
Maybe somebody has a tip here its really strange but without using `OldCollection` which is just a alias to new `Collection` class I get here:

```
TypeError: Return value of Doctrine\Tests\Common\Collections\ArrayCollectionTest::buildCollection() must be an instance of Doctrine\Common\Collections\Collection, instance of Doctrine\Collections\ArrayCollection returned
````

and

```
TypeError: Argument 1 passed to Doctrine\Tests\LazyArrayCollection::__construct() must be an instance of Doctrine\Common\Collections\Collection, instance of Doctrine\Collections\ArrayCollection given
````

I'm not sure why as `Doctrine\Common\Collections\Collection` is an alias to `Doctrine\Collections\Collection` and `ArrayCollection` is extending from it.